src/onnx/parse_randomnormal_ops.cpp

```diff
@@ -0,0 +1,82 @@
+#include <migraphx/onnx/op_parser.hpp>
+#include <migraphx/ranges.hpp>
+#include <migraphx/instruction.hpp>
+#include <migraphx/make_op.hpp>
+#include <migraphx/onnx/checks.hpp>
+#include <random>
+#include <set>
+
+namespace migraphx {
+inline namespace MIGRAPHX_INLINE_NS {
+namespace onnx {
+
+struct parse_randomnormal_ops : op_parser<parse_randomnormal_ops>
+{
+    const std::set<shape::type_t> valid_types = {
+        shape::float_type, shape::half_type, shape::double_type};
+
+    std::vector<op_desc> operators() const { return {{"RandomNormal"}, {"RandomNormalLike"}}; }
+
+    instruction_ref parse(const op_desc& opd,
+                          const onnx_parser& parser,
+                          const onnx_parser::node_info& info,
+                          std::vector<instruction_ref> args) const
+    {
+        int dtype      = 1;
+        bool use_dtype = false;
+        if(contains(info.attributes, "dtype"))
+        {
+            dtype     = info.attributes.at("dtype").i();
+            use_dtype = true;
```

Comment:
I don't think we need `use_dtype`, since the default is float and would be overriden if the attribute value is changed. For L63, we can just construct the shape using `out_type`.

Reply:
The reason I added `use_dtype` is because the default behavior for Random*Like is to use the shape and type of the input tensor, but there are two cases where you want to use only the shape but not the type: 1) the `dtype` attribute is set, and 2) the type of the input tensor is not half, single, or double precision float.

Reply:
I think I see the issue. According to the onnx spec, RandomNormal and RandomUniform have `dtype` as a mandatory attribute whereas for the others `dtype` is optional. Since we are parsing both in the same cpp file, maybe add a comment for why the boolean is necessary?